src/boot/efi/boot.c

```diff
@@ -2123,16 +2116,19 @@ static void config_load_type2_entries(
                         continue;
 
                 /* look for .osrel and .cmdline sections in the .efi binary */
-                err = pe_file_locate_sections(linux_dir, f->FileName, sections, offs, szs);
-                if (err != EFI_SUCCESS || szs[SECTION_OSREL] == 0)
+                err = pe_file_locate_sections(linux_dir, f->FileName, &sections, &nsections);
+                if (err != EFI_SUCCESS)
                         continue;
 
-                err = file_read(linux_dir, f->FileName, offs[SECTION_OSREL], szs[SECTION_OSREL], &content, NULL);
+                struct PeSectionDescriptor *section_osrel = pe_bsearch_section(sections, nsections, ".osrel");
+                if (section_osrel == NULL)
```

Comment:
coding style, we generally check for non-NULL via `if (section_osrel)`, i.e. relying on C's downgrade-to-bool feate